etc/nix/flake.nix

```diff
@@ -0,0 +1,146 @@
+    owner = "DavHau";
+    repo = "mach-nix";
+    ref = "3.1.1";
+  };
+
+  outputs = { self, nixpkgs, machnix }:
+    let
+
+      vulnerablecode-src = ./../..;
+
+      # Extract version from setup.py.
+      version = builtins.head (builtins.match ''.*version=["']?([^"',]+).*''
+        (builtins.readFile (vulnerablecode-src + "/setup.py")));
+
+      # Common shell code.
+      libSh = ./lib.sh;
+
+      # System types to support.
+      supportedSystems = [ "x86_64-linux" ];
+
+      # Helper function to generate an attrset '{ x86_64-linux = f "x86_64-linux"; ... }'.
+      forAllSystems = f:
+        nixpkgs.lib.genAttrs supportedSystems (system: f system);
+
+      # Nixpkgs instantiated for supported system types.
+      nixpkgsFor = forAllSystems (system:
+        import nixpkgs {
+          inherit system;
+          overlays = [ self.overlay ];
+        });
+
+      # mach-nix instantiated for supported system types.
+      machnixFor = forAllSystems (system:
+        import machnix {
+          pkgs = (nixpkgsFor.${system}).pkgs;
+          python = "python38";
+        });
+
+    in {
+
+      # A Nixpkgs overlay.
+      overlay = final: prev:
+        with final.pkgs; {
+
+          pythonEnv = with machnixFor.${system};
+            mkPython {
+              requirements =
+                builtins.readFile (vulnerablecode-src + "/requirements.txt");
+            };
+
+          vulnerablecode = stdenv.mkDerivation {
+            inherit version;
+            name = "vulnerablecode-${version}";
+            src = vulnerablecode-src;
+            dontConfigure = true; # do not use ./configure
+            propagatedBuildInputs = [ pythonEnv postgresql ];
+
+            postPatch = ''
+              # Make sure the pycodestyle binary in $PATH is used.
+              substituteInPlace vulnerabilities/tests/test_basics.py \
+                --replace 'join(bin_dir, "pycodestyle")' '"pycodestyle"'
+            '';
+
+            installPhase = ''
+              cp -r . $out
+            '';
+          };
+
+        };
+
+      # Provide a nix-shell env to work with vulnerablecode.
+      devShell = forAllSystems (system:
+        with nixpkgsFor.${system};
+        mkShell {
+          # will be available as env var in `nix develop` / `nix-shell`.
+          VULNERABLECODE_INSTALL_DIR = vulnerablecode;
+          buildInputs = [ vulnerablecode ];
+          shellHook = ''
+            alias vulnerablecode-manage.py=${vulnerablecode}/manage.py
+              '';
+        });
+
+      # Provide some packages for selected system types.
+      packages = forAllSystems
+        (system: { inherit (nixpkgsFor.${system}) vulnerablecode; });
+
+      # The default package for 'nix build'.
+      defaultPackage =
+        forAllSystems (system: self.packages.${system}.vulnerablecode);
+
+      # Tests run by 'nix flake check' and by Hydra.
+      checks = forAllSystems (system: {
+        inherit (self.packages.${system})
+        ;
```

Comment:
This `inherit` has no effect as is. I would suggest changing it to `inherit (self.packages.${system}) vulnerablecode;`, so that the package build itself is part of the checks. Your integration test already depends on it and so it is build in every invocation of `nix flake check` implicitly. I believe it's better to be explicit here, but if you deem it unnecessary, feel free to just remove this piece of code.

Reply:
Ah, good point. I don't know why but I somehow removed it here https://github.com/nexB/vulnerablecode/commit/ae96dcc44cceb9401545289a4d819fda3ca5c8e9#diff-f047c500dfe62ddea24cb073622316339862a843f01b655b4af33f8987a45e3bL117